### PR TITLE
E-342 fix: replace GET with POST for more secure calls

### PR DESCRIPTION
**Pull request by ifiri**
Replace GET method with POST for `auth/google` and `auth/github` calls for more security

### CODE REVIEW

**Comment by ifiri** [comment]
@dudiq your turn

**Comment by dudiq** [comment]
@ifiri why you can't test? you can change url in popup for github in dev tools console
`window.location = 'http://localhost:5000/' + window.location.search`
and see, that flow for github is working.

and what about unit tests for this methods?
maybe just add two test, that all data is passed correctly and that's all?

hmm, but maybe i'm not right about unit tests, as i can see, we don't have tests for simple methods.

**Comment by ifiri** [comment]
@dudiq endpass-auth itself have no popups or functionality. endpass-vault use iframe with link to dev server. connect-demo haven't endpass-auth in it's deps.

File: src/service/auth.js [hunk 1]
```diff
@@ -79,7 +79,9 @@ const authWithCode = async ({
  */
 const authWithGoogle = idToken =>
   request
-    .get(`${identityBaseUrl}/auth/google?token=${encodeURIComponent(idToken)}`)
+    .post(`${identityBaseUrl}/auth/google`, {
```
Comment (kazurus) [inline src/service/auth.js]: Please, add tests for these methods

**Comment by ifiri** [comment]
@dudiq tests updated
@lamartire ready for approve

File: tests/unit/specs/service/auth.spec.js [hunk 2]
```diff
@@ -152,4 +152,40 @@ describe('auth service', () => {
       });
     });
   });
+
+  describe('authWithGoogle', () => {
+    const url = `${identityBaseUrl}/auth/google`;
+
+    it('should return 200 OK', async () => {
```
Comment (dudiq) [inline tests/unit/specs/service/auth.spec.js]: what about tests with fail result?

Comment (lamartire) [reply tests/unit/specs/service/auth.spec.js]: Agree with @dudiq, it would be great to handle unsuccessful cases in tests.

**Comment by lamartire** [comment]
@kazurus, requested unit tests for failed cases. Everything else is ok

**Comment by lamartire** [comment]
@kazurus, requested unit tests for failed cases. Everything else is ok

File: tests/unit/specs/service/auth.spec.js (hunk 2 above)
Comment (ifiri) [reply tests/unit/specs/service/auth.spec.js]: @lamartire @kazurus  done.

**Pull request merged**